<launch>
	<arg name="use_ident" value="$(optenv USE_IDENTIFICATION 0)" />
  	
	<rosparam command="load"
		file="$(find labust_control)/launch/control.yaml" />
	
<!--<node pkg="labust_control" type="nu_manual" name="ManualNuRef">
	   <remap from="Enable" to="NU_enable" />
	</node>
	
	<node pkg="labust_control" type="ref_manual" name="ManualStateRef" >
	   <remap from="Enable" to="REF_enable" />
	</node> -->

	<!-- Identification control -->
<!-- <node pkg="ident_so" type="ident_so" name="identification" if="$(arg use_ident)">
	   <remap from="tauOut" to="tauIdent" />
 	 </node> -->

	<!-- Velocity controller -->
	<node pkg="labust_control" type="velcon" name="velcon">
		<remap from="out" to="tauOut" />
		<remap from="reference" to="nuRefMerged" />
		<remap from="state" to="meas_ideal" />
		<remap from="manual" to="tauManual" />
	</node>

	<!-- High-level controllers -->
<!--<group>
		<remap from="out" to="nuRef" />
		<remap from="ref" to="stateRef" />
		<remap from="state" to="stateHat" />
		<remap from="windup" to="tauAch" />
		
		<node pkg="labust_control" type="fadp_controller2" name="station_keeping">
			<remap from="Enable" to="FADP_enable" />
		</node>

		<node pkg="labust_control" type="hdg_controller2" name="hdg_control">
			<remap from="Enable" to="HDG_enable" />
		</node>

		<node pkg="labust_control" type="async_merger" name="nu_merger">
		   	<remap from="in" to="nuRef" />
			<remap from="out" to="nuRefMerged" />
			<param name="merge_nu" value="true" />
		</node>
	</group> -->
</launch>
